### PR TITLE
Move odd num_head in TransformerEncoder to slow_path

### PR DESCRIPTION
**Pull request by zrphercule**
Summary: odd nhead is not supported for masked softmax, therefore we just move it to use old slow_path

Test Plan: CI

Differential Revision: D38720086

### CODE REVIEW

**Review by erichan1 — APPROVED** [review]
Thanks for the fix for https://github.com/pytorch/pytorch/issues/83355!

**Comment by erichan1** [comment]
Failure is because we did not add odd nhead check to creating a NestedTensor. So when nhead is odd, we are sending a NestedTensor to slow path, which cannot accept NestedTensor. Best fix is probably to check if the first or any encoder layer in TransformerEncoder has odd nhead. Kind of ugly... but what can you do.
